tests/cupy_tests/creation_tests/test_from_data.py

```diff
@@ -475,24 +439,77 @@ def test_asfortranarray_cuda_array_zero_dim_dtype(
         return xp.asfortranarray(a, dtype=dtype_b)
 
 
-class DummyObjectWithCudaArrayInterface(object):
+max_cuda_array_interface_version = 2
 
-    def __init__(self, a, mask=None):
+
+@testing.gpu
+@testing.parameterize(*testing.product({
+    'ver': tuple(range(max_cuda_array_interface_version+1)),
+}))
+class TestFromExternalData(unittest.TestCase):
+    @testing.for_all_dtypes()
+    def test_asarray_cuda_array_interface(self, dtype):
+        a = testing.shaped_arange((2, 3, 4), cupy, dtype)
+        b = cupy.asarray(DummyObjectWithCudaArrayInterface(a, self.ver))
+        testing.assert_array_equal(a, b)
+
+    @testing.for_all_dtypes()
+    def test_asarray_cuda_array_interface_is_not_copied(self, dtype):
+        a = testing.shaped_arange((2, 3, 4), cupy, dtype)
+        b = cupy.asarray(DummyObjectWithCudaArrayInterface(a, self.ver))
+        a.fill(0)
+        testing.assert_array_equal(a, b)
+
+    @testing.for_all_dtypes()
+    def test_asarray_cuda_array_interface_order(self, dtype):
+        a = testing.shaped_arange((2, 3, 4), cupy, dtype)
+        b = cupy.asarray(DummyObjectWithCudaArrayInterface(a, self.ver),
+                         order='F')
+        assert b.flags.f_contiguous
+        testing.assert_array_equal(a, b)
+
+    @testing.for_all_dtypes()
+    def test_asarray_cuda_array_interface_with_strides(self, dtype):
+        a = testing.shaped_arange((2, 3, 4), cupy, dtype).T
+        b = cupy.asarray(DummyObjectWithCudaArrayInterface(a, self.ver))
+        assert a.strides == b.strides
+        assert a.nbytes == b.data.mem.size
+
+    # TODO(leofang): update this test when masked array is supported
+    @testing.for_all_dtypes()
+    def test_asarray_cuda_array_interface_with_masked_array(self, dtype):
+        if self.ver == 0:
+            raise unittest.SkipTest("Version 0 does not have the mask "
```

Comment:
Please separate parameterization as I suggested instead of using a skip.

Reply:
OK, just updated it. But why don't we just use skip here?

Reply:
We would have many skipped tests after the protocol version is incrementally increased.
I don't think that's good.
In general IMO, test skips should not be used as an alternative to easily parameterized tests.

Reply:
OK. For this case I think we only need to skip ver 0, but in general I agree with you about preferring parametrization when possible.